### PR TITLE
Draw borders

### PR DESCRIPTION
**Pull request by Chroq**
## Description

In this PR, I'm drawing borders. At the moment, I didn't make any calculation so each edge is not really well set.

Trello : [https://trello.com/c/zqtlVJDr/6-1-as-norbert-i-want-to-know-when-any-of-the-players-have-won](url)

### Game
Screenshot_20191212-091756_Expo

### CODE REVIEW

**Comment by zyhou** [comment]
Need rebase

File: board/Cell.js [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
+import React from "react";
 import {corners} from './utils';
 import {Polygon} from "react-native-svg";
 
 export const Cell = (props) => {
     const {x, y, size} = props;
     const points = corners(x, y, size);
 
-    const [color, setColor] = useState("white");
+    const color = props.player === 1 ? "#03a9fc" : props.player === 2 ? "#f4d032" : "#FFFFFF";
```
Comment (Luwangel) [inline board/Cell.js]: The ternary operator is a little bit confusing when there is `else if`. I think it's better to build a dedicated method that takes a player number and return its colour.

File: board/index.js [hunk 2]
```diff
@@ -1,18 +1,61 @@
...
 
 import {Cell} from "./Cell";
 import {gridPoints} from "./utils"
 
 export const Board = (props) => {
 
-    const grid = gridPoints(20, 75, 22, props.size, props.size).map(({props}, index) => (
-        <Cell {...props} stroke="#222222" key={index}/>
-    ));
+    const [grid, setGrid] = useState(Array(props.size * props.size).fill(0));
```
Comment (Luwangel) [inline board/index.js]: It could be simple to write a dedicated method to create "empty" grids.

File: board/index.js [hunk 3]
```diff
@@ -1,18 +1,61 @@
...
+        gridCopy[index] = player;
+        setGrid(gridCopy)
+    };
+
+    useEffect(() => {
+
+        const fetchIsWon = async (grid, player) => {
+            try {
+                const url = "https://hex-go.cleverapps.io/games/is-won";
+                const response = await fetch(url, {
```
Comment (Luwangel) [inline board/index.js]: You should move this code to a dedicated file and put the url in a config file.

Comment (Luwangel) [reply board/index.js]: ENV variables are ok to store the url.

File: training/index.js [hunk 4]
```diff
@@ -3,7 +3,10 @@ import React from "react";
 import {Board} from "../board";
 
 const TrainingScreen = () => {
-    return <Board size={6}/>
+
+    const size = 6;
```
Comment (Luwangel) [inline training/index.js]: You could pass the size as a prop, and set 6 as the default value.

File: training/index.js [hunk 5]
```diff
@@ -3,7 +3,10 @@ import React from "react";
...
+
+    return <Board size={size} mode={"training"}/>
```
Comment (Luwangel) [inline training/index.js]: You could write `mode="training"` too.

File: board/index.js [hunk 6]
```diff
@@ -1,18 +1,61 @@
...
+            }
+        };
+        fetchIsWon();
+    }, [grid]);
 
     return (
         <Svg width="500" height="500">
-            {grid}
+
+            <Polygon points="0,120 255,120 180,220" fill="#03a9fc" stroke="#222222" strokeWidth="1"/>
```
Comment (alexisjanvier) [inline board/index.js]: IMO, it could be a component in its own right

File: board/Cell.js (hunk 1 above)
Comment (Chroq) [reply board/Cell.js]: Changed by PR#6 ! Ty

File: board/index.js (hunk 3 above)
Comment (Chroq) [reply board/index.js]: As we spoke, fixed in #6 ! TY

File: board/index.js (hunk 2 above)
Comment (Chroq) [reply board/index.js]: I keep it in mind, TY

Comment (Chroq) [reply board/index.js]: We use the init of grid to generate the grid.

File: training/index.js (hunk 4 above)
Comment (Chroq) [reply training/index.js]: Done in #6 ! TY

File: board/index.js (hunk 6 above)
Comment (Chroq) [reply board/index.js]: Did it ! TY

**Pull request merged**